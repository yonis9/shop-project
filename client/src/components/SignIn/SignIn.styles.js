import styled from 'styled-components';

export const SignInContainer = styled.div`
    width: 380px;
    display: flex;
    flex-direction: column;

    @media screen and (max-width: 800px) {
        width: 300px;
        }
`;

export const SignInTitle = styled.h1`
    margin: 10px 0
`;

export const ButtonsContainer = styled.div`
    display: flex;
    justify-content: space-between;

    @media screen and (max-width: 800px) {
        flex-direction: column;
        width: 220px;
        }
`;