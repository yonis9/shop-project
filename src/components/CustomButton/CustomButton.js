import React from 'react';

import './CustomButton.scss';

const CustomButton = ({ children, isGoogleButton, ...otherProps}) => (
    <button className={`${isGoogleButton ? 'google-sign-in' : ''} custom-button`} {...otherProps}>
        {children}
    </button>
)

export default CustomButton;