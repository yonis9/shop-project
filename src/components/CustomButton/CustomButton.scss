.custom-button {
    min-width: 165px;
    width: auto;
    height: 50px;
    letter-spacing: 0.5px;
    line-height: 50px;
    padding: 0 35px 0 35px;
    font-size: 15px;
    background-color: black;
    color: white;
    text-transform: uppercase;
    font-family: 'Open Sans', sans-serif;
    font-weight: bolder;
    border: none;
    cursor: pointer;
    transition: 0.3s;
  
    &:hover {
      background-color: white;
      color: black;
      border: 1px solid black;
      transition: 0.3s;
    }
  }
  