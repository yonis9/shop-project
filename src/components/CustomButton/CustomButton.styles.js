import styled, { css } from 'styled-components';

const buttonStyles = css`
background-color: black;
color: white;
border: none;

&:hover {
    background-color: white;
    color: black;
    border: 1px solid black;
    }
`

const googleSignInStyles = css`
    background-color: #4285f4;
    color: white;

    &:hover {
        background-color: #326abc;
        border: none;
    }
`

const invertedStyles = css`
    background-color: white;
    color: black;
    border: 1px solid black;
    

    &:hover {
    background-color: black;
    color: white;
    border: none;
    }
`

const getButtonStyles = props => {
    if (props.isGoogleButton) {
        return googleSignInStyles;
    }

    return props.inverted ? invertedStyles : buttonStyles
}

export const CustomButtonContainer = styled.button`
    min-width: 165px;
    width: auto;
    height: 50px;
    letter-spacing: 0.5px;
    line-height: 50px;
    padding: 0 25px 0 25px;
    font-size: 15px;
    text-transform: uppercase;
    font-family: 'Open Sans', sans-serif;
    border: none;
    cursor: pointer;
    display: flex;
    justify-content: center;
    transition: 0.3s;

    ${getButtonStyles}

    &:focus {
    outline: 0
    }

    &:hover {
        transition: 0.3s;
    }

`