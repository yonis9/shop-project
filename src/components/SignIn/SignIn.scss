.sign-in {
    width: 410px;
    display: flex;
    flex-direction: column;

    .title {
    margin: 10px 0
    }

    .buttons {
        display: flex;
        justify-content: space-between;
    }
}



