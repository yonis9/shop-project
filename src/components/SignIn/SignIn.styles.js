import styled from 'styled-components';

export const SignInContainer = styled.div`
    width: 410px;
    display: flex;
    flex-direction: column;
`;

export const SignInTitle = styled.h1`
    margin: 10px 0
`;

export const ButtonsContainer = styled.div`
    display: flex;
    justify-content: space-between;
`;