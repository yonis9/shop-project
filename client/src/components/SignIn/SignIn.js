import React, { useState } from 'react';
import { connect } from 'react-redux';

import FormInput from  '../FormInput/FormInput';
import CustomButton from '../CustomButton/CustomButton';

import { googleSignInStart, emailSignInStart } from '../../redux/user/user-actions'

import { 
    SignInContainer,
    SignInTitle,
    ButtonsContainer
} from './SignIn.styles';

const SignIn = ({ emailSignInStart, googleSignInStart }) => {
    const [ credentials, setCredentials ] = useState({ email: '', password: '' })

    const { email, password } = credentials;

    const handleSubmit = async (e) => {
        e.preventDefault();

        emailSignInStart(email, password);


    }

    const handleChange = (e) => {
        const { name, value } = e.target;
        setCredentials({ ...credentials, [name]: value })
    }


    return(
        <SignInContainer>
            <SignInTitle>I already have an account</SignInTitle>
            <span>Sign in with your email and password</span>
            
            <form onSubmit={handleSubmit}>
                <FormInput handleChange={handleChange}
                            name='email'
                            type='email'
                            label='email'
                            value={email}
                            required />
                <FormInput handleChange={handleChange}
                            name='password'
                            label='password'
                            type='password'
                            value={password}
                            required />

                <ButtonsContainer>
                    <CustomButton type='submit'>Sign In</CustomButton>
                    <CustomButton type='button' onClick={googleSignInStart} isGoogleButton>Sign In With Google</CustomButton>
                </ButtonsContainer>
                
            </form>
        </SignInContainer>
    )
}

const mapDispatchToProps = dispatch => ({
    googleSignInStart: () => dispatch(googleSignInStart()),
    emailSignInStart: (email, password) => dispatch(emailSignInStart({ email, password }))
})

export default connect(null, mapDispatchToProps)(SignIn);