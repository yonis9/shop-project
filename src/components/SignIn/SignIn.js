import React, { Component } from 'react';

import './SignIn.scss';

import FormInput from  '../FormInput/FormInput';
import CustomButton from '../CustomButton/CustomButton';

import { signInWithGoogle } from '../../firebase/firebase';

class SignIn extends Component {
    constructor() {
        super()

        this.state = {
            email: '',
            password: ''
        }
    }

    handleSubmit = (e) => {
        e.preventDefault();

        this.setState({ email: '', password: ''})
    }

    handleChange = (e) => {
        const { name, value} = e.target;
        this.setState({[name]: value})
    }

    render() {
        const { email, password } = this.state
        return(
            <div className ='sign-in'>
                <h2>I already have an account</h2>
                <span>Sign in with your email and password</span>
                
                <form>
                    <FormInput handleChange={this.handleChange}
                               name='email' type='email'
                               label='email'
                               value={email}
                               required />
                    <FormInput handleChange={this.handleChange}
                               name='password'
                               label='password'
                               type='password'
                               value={password}
                               required />

                    <div className='buttons'>
                        <CustomButton type='submit'>Sign In</CustomButton>
                        <CustomButton onClick={signInWithGoogle} isGoogleButton>Sign In With Google</CustomButton>
                    </div>
                    
                </form>
            </div>
        )
    }
}

export default SignIn;