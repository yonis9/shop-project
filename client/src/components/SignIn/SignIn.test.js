import React from 'react';
import { shallow } from 'enzyme';

import { SignIn } from './SignIn';
import CustomButton from '../CustomButton/CustomButton';

describe('SignIn component', () => {
    let wrapper;
    let mockEmailSignInStart;
    let mockGoogleSignInStart;

    beforeEach(() => {
        mockEmailSignInStart = jest.fn();
        mockGoogleSignInStart = jest.fn();

        const mockProps = {
            emailSignInStart: mockEmailSignInStart,
            googleSignInStart: mockGoogleSignInStart
        }
        wrapper = shallow(<SignIn {...mockProps} />)
    });

    it('should render SignIn components', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call googleSignInStart when the Sign In With Google button been clicked', () => {
        wrapper.find(CustomButton).last().simulate('click');
        expect(mockGoogleSignInStart).toHaveBeenCalled();
    });


});