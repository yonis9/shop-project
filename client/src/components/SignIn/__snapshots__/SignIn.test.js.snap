// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component should render SignIn components 1`] = `ShallowWrapper {}`;
